manifests/network/dns.pp

```diff
@@ -9,9 +21,9 @@
     $_config = hiera_hash('system::network::dns', undef)
   }
   if $_config {
-    $domains     = $_config['domains']
-    $nameservers = $_config['nameservers']
-    $options     = $_config['options']
+    $domains     = pick($_config['domains'],[])
+    $nameservers = pick($_config['nameservers'],[])
+    $options     = pick($_config['options'],[])
     validate_array($domains)
```

Comment:
Unfortunately, I think these functions produce deprecation warnings these days.  I'll ask around to see what the 'correct' puppet 4+ way is.

Reply:
In Puppet 4 you can use ```assert_type(Array[String], $domains)```

Reply:
https://puppet.com/docs/puppet/5.3/function.html#asserttype is the answer.
`assert_type(Array, $domains)`
I suppose you could also do
`$domains = pick(assert_type(Optional[Array], $_config['domains']), [])` instead, but that's a bit of a mess IMO.

Reply:
@ekohl Yeah, `Array[String]` is better than just `Array`

and
`$domains = pick(assert_type(Optional[Array[String]], $_config['domains']), [])` is getting even more ugly, so I'd avoid that!